src/blockindices.jl

```diff
@@ -140,23 +140,23 @@ julia> a[BlockIndex((2,2), (2,3))]
 20
 ```
 """
-struct BlockIndex{N,TI<:Tuple{Vararg{Integer,N}},Tα<:Tuple{Vararg{Integer,N}}}
+struct BlockIndex{N,TI<:Tuple{Vararg{Any,N}},Tα<:Tuple{Vararg{Any,N}}}
```

Comment:
This generalization enables constructing indexing objects such as `Block(1)[Block(2)]` or `Block(1)[Block(2)[3]]`, for example in the case when a block array has blocks that are themselves blocked. I have use cases for that where I want to construct a block array where the blocks have extra structure and custom non-integer indexing (in particular, the blocks have a Kronecker product structure, so I want to be able to slice the blocks while preserving that structure).